spark/src/main/scala/org/apache/spark/sql/delta/stats/StatisticsCollection.scala

```diff
@@ -458,16 +458,22 @@ object StatisticsCollection extends DeltaCommand {
    * @param name The name of the data skipping column for validating data type.
    * @param dataType The data type of the data skipping column.
    * @param columnPaths The column paths of all valid fields.
+   * @param insideStruct Tracks if we are inside a struct, in which case all types are valid
    */
   private def validateDataSkippingType(
       name: String,
       dataType: DataType,
-      columnPaths: ArrayBuffer[String]): Unit = dataType match {
+      columnPaths: ArrayBuffer[String],
+      insideStruct: Boolean = false): Unit = dataType match {
     case s: StructType =>
       s.foreach { field =>
-        validateDataSkippingType(name + "." + field.name, field.dataType, columnPaths)
+        validateDataSkippingType(name + "." + field.name, field.dataType, columnPaths,
+          insideStruct = true)
       }
     case SkippingEligibleDataType(_) => columnPaths.append(name)
+    case d if insideStruct =>
```

Comment:
should we also make the non-struct field have the same behavior?

Reply:
Up to you. The idea was if you directly specify an in-eligible type, throw an exception because you did something wrong. If you specify a struct, just work with the sub fields that are supported

Reply:
Then, I am good about it.
Thank you